lib/index.js

```diff
@@ -77,6 +77,7 @@ function S3(uri, callback) {
         uri.acl = query.acl || 'private';
         uri.sse = query.sse;
         uri.sseKmsId = query.sseKmsId;
+        uri.timeout = parseInt(query.timeout);
```

Comment:
Could you add a ternary or other check here to make it clearer that when `query.timeout` is undefined `uri.timeout` ends up as a falsey value?

(reading this I had to step through the `undefined => parsed as NaN => falsey` in check below.)